### ISSUE TITLE
Cannot open the game, although nothing change in graphic drivers

### ISSUE THREAD

**Issue by vohai2003** [Anuken/Mindustry#6813]
**Platform**: *Windows*

**Build**: *7.0 Build 135.2*

**Issue**: *My laptop used to run Mindustry flawlessly, but it unable to started after update to lastest Alpha build. The error pointed out that I lacked the framebuffer_object extension. Not much things changed before this, except I installed JDK 17 for coding. My laptop supported up to OpenGL 2.1*

**Steps to reproduce**: *Just install JDK 17 and update the pre-alpha game.*

**Link(s) to mod(s) used**: *None*

**(Crash) logs**: [crash-report-05_11_2022_08_52_56.txt](https://github.com/Anuken/Mindustry/files/8666091/crash-report-05_11_2022_08_52_56.txt)

---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [X] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [X] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by Anuken** [comment]
I can't reproduce or fix this.

**Comment by vohai2003** [comment]
I tested with software rendering and it's able to started. So the problem may lie in the graphic extensions.
Just curious, is there any major OpenGL changing in this game during the pre-alpha releases?

**Comment by Anuken** [comment]
No, nothing relevant at all.

**Comment by vohai2003** [comment]
![image](https://user-images.githubusercontent.com/63694226/167796545-d7717ac2-7712-4453-9bb7-7ac4b2310e23.png)
Weird. Nothing went wrong with my graphic card. Can Java cause such a problem?

**Comment by Anuken** [comment]
I don't know; it shouldn't. I have not changed any OpenGL extension loading code at all, it still uses GLEW for that.

**Comment by vohai2003** [comment]
> I don't know; it shouldn't. I have not changed any OpenGL extension loading code at all, it still uses GLEW for that.

Ok, so tonight I gonna give it a check. Maybe the Java library for OpenGL is not working properly.

**Comment by vohai2003** [comment]
> I'm not using any existing Java libraries for OpenGL, I have my own bindings that use GLEW.

Is there any tools that can help me debug by mind own? I'm quite new to Java.

**Comment by vohai2003** [comment]
> I don't think so. Frankly, I don't even know how I would debug this myself, even if I had access to your computer. GLEW just isn't finding the extension, and as I didn't write that library, I don't know why.

OK thanks.
So I thought I've found sth that may cause this problem.
Is Mindustry require GL_ARB_framebuffer_object? My graphic card don't support that extension but do support GL_EXT_framebuffer_object.

**Comment by vohai2003** [comment]
> It should be loading the EXT version - that is what I check for.

Ok, so after several tests, I've come back here with some infos I found.
1. I used Mesa3D for testing around with extension lacking, and confirm that the game only requires EXR version as you said.
2. Also using Mesa3D, I realized that the error message for the test case with Mesa and without Mesa is really different.
With Mesa, the error message is "Make sure you have OpenGL 2.0 and FBO"; as the exception caught by the game.
Without Mesa, the message is "missing framebuffer object extension"; thrown by SDL.
3. Right now I cant play any version of Mindustry, both old and new. Can Java cause all of these problem? I used to run Mindustry on JVM 8 until I replace it with JDK 17.
